doc/test.md

```diff
@@ -55,6 +59,29 @@ Then, to run the tests:
 ./node_modules/mocha/bin/mocha test_csi.js
 ```
 
+## Using PCIe NVMe devices in cargo tests while developing
+
+When developing new features, testing those with real PCIe devices in the process might come in handy.
+In order to do so, the PCIe device first needs to be bound to the vfio driver:
+
+```bash
+sudo PCI_ALLOWED="<PCI-ADDRESS>" ./spdk-rs/spdk/scripts/setup.sh
+```
+
+The bdev name in the cargo test case can then follow the PCIe URI pattern:
+
+```rust
+static BDEVNAME1: &str = "pcie:///<PCI-ADDRESS>";
+```
+
+After testing the device may be rebound to the NVMe driver:
+
+```bash
+sudo PCI_ALLOWED="<PCI-ADDRESS>" ./spdk-rs/spdk/scripts/setup.sh reset
+```
+
+Please do not submit pull requests with active cargo test cases that require PCIe devices to be present.
```

Comment:
Do you mean a rust feature flag, such that PCIe support can be selected at compile time?
It makes sense to me! I briefly thought about adding a feature flag because of the failing test case but went with the `ignore` attribute for simplicity.  
I can rework this to be a rust feature flag within this PR, however, if I attempt this (https://doc.rust-lang.org/cargo/reference/features.html) I get the following error:
```
error: failed to parse manifest at `/home/dennis/src/mayastor/Cargo.toml`

Caused by:
  this virtual manifest specifies a `features` section, which is not allowed
```
Do you know what needs to be done here?

A QEMU NVMe emulated device via PCIe would work for testing. One just needs to make sure that the cache device is configured with `ms=8` https://qemu-project.gitlab.io/qemu/system/devices/nvme.html#metadata.

Reply:
> Do you know what needs to be done here?

The feature flag would need to be added to the io-engine crate, see some examples: https://github.com/openebs/mayastor/blob/2825e5eb8042999718c6017229385b731f9b9b5a/io-engine/Cargo.toml#L12

I'll leave it up to you, fine to keep ignore for now as well.
<br>

> A QEMU NVMe emulated device via PCIe would work for testing. One just needs to make sure that the cache device is configured with ms=8 https://qemu-project.gitlab.io/qemu/system/devices/nvme.html#metadata.

Maybe something we may be able to do here then. We run the python BDD tests in a vm, maybe we can configure it with an emulated PCIe device and be able to run these tests then?
Alternatively we can setup VM's on demand from the tests themselves? With nix we can setup test vm's easily, example: https://github.com/openebs/mayastor-extensions/blob/develop/tests/helm/test.nix 
But I don't think there's emulated nvme support there...

Reply:
I now introduced a new feature flag :)


Yes we should definitely look into running the test in an emulated QEMU environment or something similar.

That on demand VM setup you linked looks interesting. It would take a bit time to play with it and figure out the capabilities.
Should we address the CI setup within this PR?

Reply:
We can do in another PR, let's get this one merged :)

Reply:
I will talk with you outside of this PR to discuss this :)